lib/rules/indent.js

```diff
@@ -442,6 +442,7 @@ class OffsetStorage {
                 const offset = (
                     offsetInfo.from &&
                     offsetInfo.from.loc.start.line === token.loc.start.line &&
+                    !/^\s*?\n/.test(token.value) &&
```

Comment:
This addition was necessary due to JSX behavior. When parsing code like this:

```js
<a>
    <b>
    </b>
</a>
```

There is a hidden `JSXText` (or `Literal`) token between `<a>` and `<b>`, containing a newline followed by whitespace. The `indent` rule had previously been relying on the old template literal behavior to ensure that the `<b>` tag would get indented properly.

In other words, the `indent` rule would previously indent the above code in effectively the same manner as the below code:

```js
a(`
    `, { // (this object is like the `b` tag)
    }, `
`);
```

However, by fixing this bug it would no longer possible to treat the two pieces of code in the same way anymore. The fix is to modify the token collapsing behavior (which checks whether two tokens are on the same line) to consider two tokens to be on separate lines if the second token starts with a newline character.